src/Knoop/Node/src/put.hpp

```diff
@@ -0,0 +1,11 @@
+template< typename Str, typename T >
```

Comment:
Why `Str` here instead of `std::string`?

Reply:
consider the following example:
```cpp
#include <iostream>

void print(const std::string& s){
  std::cout << s << std::endl;
}

int main(){
  print("hello world");
}
```

In this example, a temporary string is constructed and destroyed as part of the `print` invocation. This is unnecessary as C-style character string has a ostream overload. Strings require a heap allocation (setting aside the small buffer optimization for a moment), and as such, this is not an insignificant cost.  

We can rewrite `print` to avoid the temporary variable using a template.

```cpp
template<typename T>
void print(T&& s){
  std::cout << s << std::endl;
}
```

Here we use a forwarding reference to preserve const-ness, reference-ness, and violatile-ness of the argument.

This change does something similar in the `put` method. Previously, the signature required the construction of a temporary strings for a common use case. This version does not.